### DISCUSSION TITLE
Returning Numerous Rows When One Expected

### DISCUSSION THREAD

**Discussion by CallumInsight** [rappasoft/laravel-livewire-tables#1170] [General]
I have this relationship on my Account model

```
public function lastLoginUser(): HasOne 
{ 
 return $this->hasOne(User::class)->ofMany('last_login_at', 'max'); 
}
```

When I add this column to my data tables columns method I am returned four rows when the raw sql query only returns one row.

```
protected $model = Account::class;

Column::make("Last Login", "lastLoginUser.last_login_at")
                ->sortable(),
```

Can anyone see what I am doing wrong?

**Comment by lrljoe** [comment]
One "ofMany" issue has been a problem before, see issue #1005, it's something I'm fully intending to look at in the not too distant future.

I've given an example for an interim fix (surprisingly of almost exactly the same issue) here, which should get it working for you at least!

[https://github.com/lrljoe/livewire-tables-latestofmanyissue/blob/Fixed-Issue/app/Http/Livewire/UserTable.php](https://github.com/lrljoe/livewire-tables-latestofmanyissue/blob/Fixed-Issue/app/Http/Livewire/UserTable.php)

**Reply by CallumInsight** [reply]
Thanks Joe appreciate your reply and solution.

**Reply by lrljoe** [reply]
No problem, I have noticed that the example doesn't work smoothly with the column select.

 But you can use the customSlug method on the column to get around that.

If you get stuck let me know and I'll update that example to show what needs doing.

**Comment by koolsb** [comment]
I'm running into this same issue. I'm able to get the column to display with the solution shared as well as with a label column, however I cannot sort off of this column since the value doesn't match the actual displayed field. Joining the tables brings back the duplicate records even though I just want the latest of many in this instance.

**Reply by lrljoe** [reply]
When using a label, you need to use a callback in the sortable, see the second example here:
[https://rappasoft.com/docs/laravel-livewire-tables/v3/columns/available-methods#content-sorting](https://rappasoft.com/docs/laravel-livewire-tables/v3/columns/available-methods#content-sorting)

E.g.
```
Column::make(__('Address Line 1'))
    ->sortable(
        fn(Builder $query, string $direction) => $query->orderBy('address.line1')
    ),
```
